### ISSUE TITLE
Bypassing Defender

### ISSUE THREAD

**Issue by notevil1337** [UnamSanctam/SilentCryptoMiner#1852]
Hi Unam, I thought this program could bypass windows defender and it doesn't?
Or it needs special settings for this? I am using your default settings

**Comment by UnamSanctam** [comment]
The "Add Defender Exclusions" adds exclusions to make it not detected after it has been started the first time, but the initial filetime scan of the program won't be undetected usually. It's a public open source program so it commonly gets detected a few days after each release (to be fair it's usually the same for private paid ones too).

**Comment by notevil1337** [comment]
Well, if I hack into the system with my code and disable defender from the registry, if that worked and the virus didn't catch it, logically the antivirus wouldn't detect the miner right? I will add the downloader to the script itself.
And also tell me is it possible to change your open source code and stop some anti-viruses (WD at least) or is it a waste of time and changing all functions and variables will not lead to results?

**Comment by notevil1337** [comment]
Can UnamBinder bypass the defender ?

**Comment by notevil1337** [comment]
I also have problems in visual studio when I open forms it throws me this errors
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/f65415df-78cc-4360-8fc4-f8982fa84a09)

**Comment by notevil1337** [comment]
@UnamSanctam You gone?

**Comment by UnamSanctam** [comment]
>Well, if I hack into the system with my code and disable defender from the registry, if that worked and the virus didn't catch it, logically the antivirus wouldn't detect the miner right? I will add the downloader to the script itself.
And also tell me is it possible to change your open source code and stop some anti-viruses (WD at least) or is it a waste of time and changing all functions and variables will not lead to results?

Yes if you "disable" Defender first then it should not catch it. You can change the code to reduce the detections but it's quite complicated work.

>I also have problems in visual studio when I open forms it throws me this errors

Try building the miner first, try re-downloading the miner project and then build it before opening any forms.

**Comment by notevil1337** [comment]
It worked bro

**Comment by notevil1337** [comment]
When I downloaded the open source, Windows Defender detected the miner builder itself, after that I changed all the function names in the codedom class and Windows Defender didn't detect it as a virus anymore.

What exactly should I change in the code to make it less detectable, functions, variables or the logic of the code itself (which will be a bit tricky, regardless of the fact that I'm a C# dev) I also want to change the stub itself, but I couldn't find where to find the file

**Comment by notevil1337** [comment]
@UnamSanctam

**Comment by notevil1337** [comment]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/961ce79f-f930-4fc1-8927-ef9d5723d8bb)
Also I got that error while building the miner, without any changes in the code

**Comment by notevil1337** [comment]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/85edbdf9-c901-4d72-942c-b8468d7d40fd)
Are they supposed to be like that?

**Comment by UnamSanctam** [comment]
>Also I got that error while building the miner, without any changes in the code

Hmm, the builder should work after just building it, I tried it a few days ago (when I was helping another issue). Can you try building the builder as "Release" if you aren't already?

>Are they supposed to be like that?

Sure that doesn't matter.

>What exactly should I change in the code to make it less detectable, functions, variables or the logic of the code itself (which will be a bit tricky, regardless of the fact that I'm a C# dev) I also want to change the stub itself, but I couldn't find where to find the file

All the files that the miner is comprised of is inside this folder https://github.com/UnamSanctam/SilentCryptoMiner/tree/master/SilentCryptoMiner/Resources/Code, the `miner.cpp` is the main miner file and `watchdog.cpp` is the watchdog. Then inside the `Includes` folder are the other files used by the miner and the watchdog. Then there is also the syscalls files that are generated by SysWhispersU during compilation. To reduce detection you'll have to find what causes them and then change them. Either you can go by a purely trial and error approach and just remove a little code at a time and then test it inside a VM to see if it gets detected or not. If it stops getting detected then you've probably found something that causes a detection. Or you can do the inverse and start with most code commented out and then uncomment a little at a time until a detection appears. Or you can make use of some kind of scanning tool like CAPA to find if something suspicious is getting detected inside the program.

**Comment by notevil1337** [comment]
Yes I already building the builder as "Release"
As I understand it, changing the code in C# itself will not help to avoid detection? Should I change only the CPP files (of the main miner and watchdog)?
Or changing/modifying the C# code itself like 
(CodeDom) By changing Function names, deleting some part of the code, changing file names itself and so on

**Comment by UnamSanctam** [comment]
The miner that gets compiled is only in C++, the C# builder is only for building the C++ miner. By modifying the C# you would make the builder undetected but the miner would stay the same. So modifying the C++ code is what you'd do to reduce miner detections.

**Comment by notevil1337** [comment]
Ok thanks

**Comment by notevil1337** [comment]
Ill try..

**Comment by notevil1337** [comment]
Oh and i cant fix that error
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/409fe622-4f97-48f8-980b-f1d437b15b62)

**Comment by UnamSanctam** [comment]
Hmm, that one is a bit difficult to solve because it doesn't say much. You can try running it in debug from within Visual Studio and see if it breaks it where there error occurs, though I don't think that will happen. The error is happening within workerBuild_DoWork, can you try adding the project folder into your Windows Defender exclusions (or whatever antivirus you have)? Your antivirus could be removing some of the resource files or something (even when the antivirus is disabled it can remove things).

**Comment by notevil1337** [comment]
Sorry I did not understand (Windows Defender exclusions), How can I add the project folder into it

**Comment by UnamSanctam** [comment]
Inside your Windows Defender settings you can find "Exclusions" near the bottom where you can add the folder that you have the project in.

**Comment by notevil1337** [comment]
Also I have a project open on my own pc and not in a VirtualBox

**Comment by notevil1337** [comment]
Oh okay lemme try that

**Comment by notevil1337** [comment]
Will I have problems adding the project to Exclusions if I'm on my computer?

**Comment by UnamSanctam** [comment]
No, when I work I have all project files under exclusions. Though of course it's not too good to do that for any files you don't trust enough.

**Comment by notevil1337** [comment]
Ok I add project to exclusions

**Comment by notevil1337** [comment]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/2113e194-62bf-42fc-a2b3-3af6950e168b)
Found something in a Debug Mode

**Comment by notevil1337** [comment]
Yep I got only that error in a debug. 
string injectionTarget = FormAO.toggleRootkit.Checked ? "dialer.exe" : Invoke(new Func<string>(() => miner.comboInjection.Text)).ToString();

**Comment by UnamSanctam** [comment]
Did you change the .NET version of the project or something? Or did you remove something from the forms?

**Comment by notevil1337** [comment]
BRO I am so sorry lmao
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/63879c21-e9f3-49bd-9b77-6921aa7e4e7c)
When I opened the project for the first time, when I had a problem opening the forms, at that time I opened the advanced form first, when I got an error, then I just clicked continue

**Comment by notevil1337** [comment]
But I updated (downloaded) the project 2 times, it didn't help as I see

**Comment by UnamSanctam** [comment]
Aha yeah then that's why, the control that it's trying to read doesn't exist anymore. It's best to delete the files, download them again and then when you open the project build it before anything else so that it generates all the files.

**Comment by notevil1337** [comment]
Alright! I think you would be the greatest teacher ever lol.

**Comment by notevil1337** [comment]
Your answers are very clear.

**Comment by notevil1337** [comment]
I'm sorry if I bother you again @UnamSanctam 
In order for the miner to work successfully/perfectly/permanently, is it necessary to enable watchdog?
Or in general, tell me which settings are necessary for the miner to work properly. I am just testing things.. 
thanks

**Comment by UnamSanctam** [comment]
Yes then the watchdog is recommended, since it restarts the miner if killed or resinstalls it if it's removed.

**Comment by notevil1337** [comment]
@UnamSanctam 
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/15dc259f-c936-4da6-9a9a-aecc3f71e8fd)
Why this happens? 
And something else I noticed when I copy my Monero(XMR) address it pastes a different address every time in the virtual box. 
Then I wondered why it was running like that (if the virtual machine was infected with a hack-paste code) and I installed the miner on my computer, then I ran it on the virtual machine and at the same time it threw this error on the panel (when I used my real address).
In short, after the error and after the build/compilation, when I opened the builder again on my machine and added the Monero(XMR) address, it did the same thing again and added another address instead of mine lol.
I use GUI wallet for XMR.

**Comment by UnamSanctam** [comment]
It also copied someone else's address on your real machine? If it's pasting another crypto address then the one you copied then you most likely have a crypto clipper on your computer. Try downloading another antivirus like Avast and scan your computer with it. If it cannot connect when you use your real address then there's probably something wrong with it, try running the checker that got built together with your miner to see what it says.

**Comment by notevil1337** [comment]
It is weird af
I think it is fixed on my computer.
In vm its really weird, I made a second address and it copy-pastes normally
My first address 45CG5UFcRgqPEZDNQD3iuoV1c5nqg8BEpYamGXFcDf55g8kN81AWWJHVJAPNsHCn7uAfXgmDNr4DGjeZPz7xEUZz8rFVuno
and it literally pastes this 42hejQfEyTdXCNjRDFbniQFmFEVS6Q21hjZ7kAq3HAcp3eJYaLmuNNnj83HU52bVnaY48hhPgzWmLBkTxDYLVzPnGWn463k

**Comment by notevil1337** [comment]
Also why the second one's length is smaller

**Comment by UnamSanctam** [comment]
They are the same length (the font just gives them different widths), but is it only happening in your VM or your normal computer as well? If it's only the VM then it's not too bad since you can always reset it easily.

**Comment by notevil1337** [comment]
Yeah i think my pc is good now
Look at this, Its not even me lmfao what the hell 
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/60a47ee5-d849-4707-8085-5bfdbe4f2c65)

**Comment by notevil1337** [comment]
Oh shit I know... fucking VirusTotal i uploaded there only one time

**Comment by UnamSanctam** [comment]
Haha yeah the username looks like a sandbox (like VirusTotal).

**Comment by notevil1337** [comment]
Yep
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/aa7b4809-639a-4a4b-8da3-082bf4201187)
I think it works now with my real address lol, I got no error because I turn Watchdog On

**Comment by godrozhdenia** [comment]
> When I downloaded the open source, Windows Defender detected the miner builder itself, after that I changed all the function names in the codedom class and Windows Defender didn't detect it as a virus anymore.
> 
Hello, do you mind sharing the code to bypass Defender detecting the builder?
